pkg/controller/controller.go

```diff
@@ -387,6 +387,11 @@ func (c *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVolu
 		return nil, status.Errorf(codes.NotFound, "could not retreive volume [%s]: %v", vID, err)
 	}
 
+	if vol.Status.StagingPath != "" || vol.Status.ContainerPath != "" {
```

Comment:
Do not proceed deletion if the volume hasn't been unpublished and unstaged. Repond with proper error code (codes.FailedPrecondition) if the DeleteVolume RPC was prematurely called.

Reply:
It is better to check status instead of weak signals such as StagingPath and ContainerPath for staging and publishing.